pkg/actions/karpenter/create.go

```diff
@@ -73,6 +73,8 @@ func (i *Installer) Create() error {
 		return fmt.Errorf("failed to save the identity config: %w", err)
 	}
 
+	instanceProfileArn := fmt.Sprintf("arn:aws:iam::%s:instance-profile/eksctl-%s-%s", parsedARN.AccountID, builder.KarpenterNodeInstanceProfile, i.Config.Metadata.Name)
+
 	// Install Karpenter
-	return i.KarpenterInstaller.Install(context.Background(), roleARN)
+	return i.KarpenterInstaller.Install(context.Background(), roleARN, instanceProfileArn)
```

Comment:
Okay, so this is a cool approach. However, what would be better is to have this as a configuration value in the config file surfaced. Which means users can set it, BUT it needs a default value of the generated one.

So leave this as is, have an optional field of type string and if it IS defined, then just overwrite this value.

But also, pass it through as you do before but instead of a boolean, just always set the value and always provide this field in the helmchart since we never want to NOT provide it because `eksctl` is using a custom value.